error: ‘max_length’ may be used uninitialized in this function [-Werror=maybe-uninitialized]

With ucx-1.21.0,  I saw a compile issue.  I use gcc-11.4.

```
proto/proto_select.c: In function ‘ucp_proto_select_elem_init’:
proto/proto_select.c:389:25: error: ‘max_length’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  389 |     } while (max_length < SIZE_MAX);
      |                         ^
proto/proto_select.c:340:24: note: ‘max_length’ was declared here
  340 |     size_t msg_length, max_length;
      |                        ^~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [Makefile:1371: proto/libucp_la-proto_select.lo] Error 1
```

--- a/src/ucp/proto/proto_select.c
+++ b/src/ucp/proto/proto_select.c
@@ -360,6 +360,7 @@ ucp_proto_select_elem_init_thresh(ucp_worker_h worker,
      * possible message sizes until SIZE_MAX.
      */
     msg_length = 0;
+    max_length = SIZE_MAX;
     do {
         ucs_array_init_dynamic(&perf_list);
         ucs_array_init_dynamic(&envelope);

